### ISSUE TITLE
Keycloak integration

### ISSUE THREAD

**Issue by flavienlaurent** [kumuluz/kumuluzee-swagger#4]
I'm trying to use Swagger & Swagger UI extension. It works well with JAX-RS API but it doesn't take care of security.

I'm using the Keycloak extension but the Swagger extension (and the `JaxRsSwaggerAnnotationProcessor` class) doesn't seem to handle it properly. Also, even if I add some security definitions in the `@SwaggerDefinition` like this:

```
@SwaggerDefinition(
  info = @Info(title = "foobar", version = "v1.0.0"),
  host = "localhost:8081",
  securityDefinition =
      @SecurityDefinition(
        oAuth2Definitions =
            @OAuth2Definition(
              key = "foo",
              flow = Flow.PASSWORD,
              tokenUrl = "http://foo.com",
              description = "bar"
            )
      )
)
```
the swagger.json file doesn't change.

Thanks in advance!

**Comment by zvonegit** [comment]
I tested the `@SecurityDefinition` on `kumuluzee-swagger` sample project and the swagger.json is updated correctly:

```java
@SwaggerDefinition(info = @Info(title = "CustomersAPI", version = "v1.0.0"), host = "localhost:8080", securityDefinition =
@SecurityDefinition(
        oAuth2Definitions =
        @OAuth2Definition(
                key = "foo",
                flow = OAuth2Definition.Flow.IMPLICIT,
                tokenUrl = "https://test-orcl.vm.apimanager.cloud.si/auth/realms/kumuluz-api/protocol/openid-connect/token",
                authorizationUrl = "https://test-orcl.vm.apimanager.cloud.si/auth/realms/kumuluz-api/protocol/openid-connect/auth",
                description = "Keycloak Auth",
                scopes = {}
        )
))
```

and the resulting swagger.json is:

```xml
...
"securityDefinitions" : {
    "foo" : {
      "description" : "Keycloak Auth",
      "type" : "oauth2",
      "authorizationUrl" : "https://test-orcl.vm.apimanager.cloud.si/auth/realms/kumuluz-api/protocol/openid-connect/auth",
      "flow" : "implicit"
    }
  },
...
```

Also, the Swagger UI should provide Authorize button when Security definition is defined and provided.

Regarding the Keycloak extension, the integration with the Swagger extension is not yet provided. We will provide this in the next version of the extension.

**Comment by flavienlaurent** [comment]
Thanks for your response. I think I found something weird that may cause my issue.

If you put the class annoted with `@SwaggerDefinition` (ie. application class) in the same package of the `@Api` annoted classes (ie. resource classes), it works fine because the `swagger-configuration.json` file is generated with the right `resourcePackages` value (which is very important for the security definitions parsing.

But try to separated resources in another package,

![image](https://user-images.githubusercontent.com/4429434/42727127-29231eaa-87a1-11e8-850c-319ddfb59db5.png)

the security definitions are not parsed anymore because the `@SwaggerDefinition` annoted class is never read by `io.swagger.jaxrs.Reader` (see `com.kumuluz.ee.swagger.servlets.ApiListingServlet#L62` then `io.swagger.jaxrs.Reader#L130-167`).

**Comment by zvonegit** [comment]
Fixed in v1.2.0
